infra/mock_api/template/ecs.tf

```diff
@@ -109,21 +106,132 @@ resource "aws_ecs_task_definition" "mock-api-ecs-task-definition" {
         {
           containerPort : 8080
         }
-      ],
+      ]
+      readonlyRootFilesystem = true
+      linuxParameters = {
+        capabilities = {
+          drop = ["ALL"]
+        },
+        initProcessEnabled = true
+      }
+      environment : [
+        {
+          "name" : "POSTGRES_USER"
+          "value" : "${data.aws_ssm_parameter.db_username.value}"
+        },
+        {
+          "name" : "POSTGRES_PASSWORD"
+          "value" : "${data.aws_ssm_parameter.db_pw.value}"
+        },
+        {
+          "name" : "POSTGRES_DB"
+          "value" : "${data.aws_ssm_parameter.db.value}"
```

Comment:
You probably want to set this to `${aws_db_instance.mock_api_db.db_name}`? I see you hardcoded the db_name to `main` when creating the RDS instance instead of passing in the SSM parameter value.